orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+                                    "%s:%d: fi_setopt failed: %s\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret));
+                free_conduit_resources(cur_conduit);
+                /* abort this current transport, but check if next transport can be opened */
+                  continue;
+            }
+
+            ret = fi_recv(orte_rml_ofi.ofi_conduits[cur_conduit].ep,
+                          orte_rml_ofi.ofi_conduits[cur_conduit].rxbuf,
+                          orte_rml_ofi.ofi_conduits[cur_conduit].rxbuf_size,
+                          fi_mr_desc(orte_rml_ofi.ofi_conduits[cur_conduit].mr_multi_recv),
+                          0,&orte_rml_ofi.ofi_conduits[cur_conduit].rx_ctx1);  
+            if (ret) {
+                        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                                    "%s:%d: fi_recv failed: %s\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret));
+                free_conduit_resources(cur_conduit);
+                /* abort this current transport, but check if next transport can be opened */
+                  continue;
+            }
+            /**
+            *  get the fd  and register the progress fn
+            **/
+            ret = fi_control(&orte_rml_ofi.ofi_conduits[cur_conduit].cq->fid, FI_GETWAIT,
+                            (void *) &orte_rml_ofi.ofi_conduits[cur_conduit].fd);
+            if (0 != ret) {
+                opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                                    "%s:%d: fi_control failed to get fd: %s\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret));
+                free_conduit_resources(cur_conduit);
+                /* abort this current transport, but check if next transport can be opened */
+                  continue;
+            }
+
+            /* - create the event  that will wait on the fd*/
+            /* use the opal_event_set to do a libevent set on the fd
+            *  so when something is available to read, the cq_porgress_handler
+            *  will be called */
+            opal_event_set(orte_event_base,
+                       &orte_rml_ofi.ofi_conduits[cur_conduit].progress_event,
+                       orte_rml_ofi.ofi_conduits[cur_conduit].fd,
+                       OPAL_EV_READ|OPAL_EV_PERSIST,
+                       cq_progress_handler,
+                       &orte_rml_ofi.ofi_conduits[cur_conduit]);
+            opal_event_add(&orte_rml_ofi.ofi_conduits[cur_conduit].progress_event, 0);
+            orte_rml_ofi.ofi_conduits[cur_conduit].progress_ev_active = true;
+           /** update the number of conduits in the ofi_conduits[] array **/
+           opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                 "%s:%d Conduit id - %d created ",__FILE__,__LINE__,orte_rml_ofi.conduit_open_num);
+           orte_rml_ofi.conduit_open_num++;
```

Comment:
Is there a bounds checking for `MAX_CONDUIT` here somewhere?

With the layered providers that are coming, it's quite feasible that there can be more than 20 providers that come back from `fi_getinfo()`.